### PR TITLE
Inject runtime info into v8env

### PR DESCRIPTION
**Pull request by michaeldwan**
I wanted a way to check the version of v8env being ran, so I made a rollup plugin that grabs the build number from a shell script and injects it into `fly.runtime.build`. `build.rs` is also using this script to get the build number used in the Server header. It's the same as before in CI, `{git_sha[0..8]}`, but `{git_sha[0..8]}-dev` elsewhere.

### CODE REVIEW

File: v8env/src/globals.ts [hunk 1]
```diff
@@ -111,6 +112,7 @@ window.fly = {
   data: flyData,
   http: flyHttp,
   Image: Image,
+  runtime,
```
Comment (jeromegn) [inline v8env/src/globals.ts]: I think you need to add that to the fly interface nested in the window interface above.

File: v8env/rollup.config.js [hunk 2]
```diff
@@ -67,6 +68,28 @@ function strings(
...
+
+  return {
+    name: "runtimeInfo",
+    transform: (code, id) => {
+      if (filter(id)) {
+        const build = execSync('./scripts/build-number.sh', {
+          cwd: '..'
+        }).toString().trim();
+
+        const magicString = new MagicString(`export const runtime = { build: "${build}" };`);
```
Comment (jeromegn) [inline v8env/rollup.config.js]: This is an interesting package. I was gonna suggest https://github.com/rollup/rollup-plugin-replace, but it appears to be using magic-string itself!

File: scripts/build-number.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+version=$(git rev-parse HEAD | cut -c-8)
+
+if [ -z "$CI" ]; then
+  version="$version-dev"
+fi
+
+echo $version
```
Comment (jeromegn) [inline scripts/build-number.sh]: Probably nothing, but echo might output a trailing new line there. Just to be safe `echo -n` will not output a new line. I don't know how that's handled from rust and js though, so might be nothing.

Comment (michaeldwan) [reply scripts/build-number.sh]: good call! Rust does the right thing but js needed `trim`. I just tried `echo -n` and it's working when called directly not from the script, but I just realized this is what `printf` is for! 😄

**Pull request merged**